### PR TITLE
Lua for sile

### PR DESCRIPTION
**Pull request by teto**
###### Motivation for this change
forgot to add cassowary in the previous PR. 
Also tried to move pulseaudio to the conventional package. @doronbehar could you check I haven't broken the package, there is a warning aobut a missing  folder

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by doronbehar** [comment]
@teto along with fixing the conflict, there's an interesting package I think we could put in `luarocks-packages.csv`: https://github.com/Koihik/LuaFormatter

Could you please try to add it and I'll test it?

File: pkgs/development/lua-modules/overrides.nix [hunk 1]
```diff
@@ -294,6 +294,9 @@ with super;
     };
   });
 
+  pulseaudio = super.pulseaudio.override({
+    nativeBuildInputs = [ pkgs.pulseaudio pkgs.pkg-config ];
```
[the remark this answers is not in the document]
Comment (teto) [reply pkgs/development/lua-modules/overrides.nix]: seems like the luarocks bump broke some things for me as it stopped building with `Warning: unmatched variable LUA_LIBDIR` (and it used to at least build). The makeFlags should not be necessary as the lua infra should take care of it

[the remark this answers is not in the document]
Comment (doronbehar) [reply pkgs/development/lua-modules/overrides.nix]: Check out my comment here: https://github.com/luarocks/luarocks/issues/1155#issuecomment-585749759 . I've tested compilation of libluv on Nixpkgs with Luarocks 3.2.1 and I still had the same warnings and errors, it's just that they haven't broke the build. Waiting for upstream to respond.

**Comment by teto** [comment]
I've had a go at the package but it seems huge (42 MB), to compile tons of stuff with some difficulties.
```
    file cannot create directory: /var/empty/local/bin.  Maybe need
    administrative privileges.
```
Sorry I do not have time for this at the moment but I might give it a try once it goes out of `dev` mode.

**Comment by doronbehar** [comment]
O.K @teto no big deal. In general, what is the best way to make a PR with a new Luarock based package? I once tried to add a line in `luarocks-packages.csv` for a package I wanted to add and I ran the update script but it updated so many more packages that I hadn't had the nerve to put all of these changes in my PR.

**Comment by teto** [comment]
@doronbehar that's how you describe. Just at the end don't commit everything but use `git add -p` to only commit the changes you are interested. You can discard the rest via `git reset --hard` or `git stash`

**Comment by doronbehar** [comment]
I see. And what if we'd like to update `lua.pkgs.libluv` for the sake of _maybe_ fixing #79870, how would you go and do that?

**Pull request closed without merging**